qcodes/utils/validators.py

```diff
@@ -625,19 +641,111 @@ def validate(self, value: Any, context: str = '') -> None:
         for v in self._validators:
             try:
                 v.validate(value, context)
-                return
+                if self._combiner == 'OR':
+                    return
             except Exception as e:
                 # collect the args from all validators so you can see why
-                # each one failed
+                # each one that was tested failed
                 args = args + list(e.args)
+                if self._combiner == 'AND':
+                    raise ValueError(*args)
 
-        raise ValueError(*args)
+        if self._combiner == 'OR':
+            raise ValueError(*args)
 
     def __repr__(self) -> str:
         parts = (repr(v)[1:-1] for v in self._validators)
         return '<MultiType: {}>'.format(', '.join(parts))
 
 
+class MultiTypeOr(MultiType):
+    """
+    Allow the combination of several different validators.
+    The resulting validator acts as a logical OR between the
+    different validators.
+
+    Example:
+        To allow numbers as well as "off":
+            MultiTypeOr(Numbers(), Enum("off"))
+
+    Raises:
+        TypeError: If no validators provided. Or if any of the provided
+            argument is not a valid validator.
+    """
+
+    def __init__(self,
+                 *validators: Validator[Any],
+                 ) -> None:
+        super().__init__(*validators, combiner='OR')
+        if not validators:
+            raise TypeError('MultiType needs at least one Validator')
+
+        for v in validators:
+            if not isinstance(v, Validator):
+                raise TypeError('each argument must be a Validator')
+
+            if v.is_numeric:
+                # if ANY of the contained validators is numeric,
+                # the MultiType is considered numeric too.
+                # this could cause problems if you want to sweep
+                # from a non-numeric to a numeric value, so we
+                # need to be careful about this in the sweep code
+                self.is_numeric = True
+
+        self._validators = tuple(validators)
+        self._valid_values = tuple(vval for v in self._validators
+                                   for vval in v._valid_values)
```

Comment:
It looks like most of this code is duplicated from the base class and could be removed?

Reply:
Fixed.